src/file_parsing.jl

```diff
@@ -253,6 +253,35 @@ get_parameter_values(
     names::Union{NAMESTYPE, AbstractString},
 ) = get_parameter_values!(pd, names, nothing)
 
+
+"""
+    merge_toml_files(filepaths...)
+
+Parses and merges all of the given TOML filepaths and stores them in a temporary file.
+This allows a toml_dict to be constructed from multiple TOML files.
+Warns when there are non-unique TOML entries.
+"""
+function merge_toml_files(filepaths...)
```

Comment:
AI like the idea of adding a merge functionality. There is `Base.merge`, and arguably we should subtype `AbstractTOMLDict` by `AbstractDict` and just overload `Base.merge` and `Base.merge!`. Thoughts? If so I'd probably get rid of the `open(f-> TOML.print(f, merged_toml), output_file, "w")` below. Also, we might be able to do better with initializing the merged toml dict. Why does `merge_toml_files` return a file?

Reply:
It's meant to return a filepath to pass into `create_toml_dict` as an override, rather than initialize a proper `AbstractTOMLDict`. I figured this would be easier because it doesn't change the API. I think @odunbar had some reasons for this choice as well

Reply:
Is there a reason we need to create a merged toml file? I figured that we would read the two toml files in and create a merged TomlDict in memory. Or is this needed by the UQ team?

Reply:
Related to Charlie's point about actually writing a new TOML. I don't mind whether we write a new TOML, or just have a dict. Perhaps we can do both, we always output a dict, and if the user requests it via a flag we can write a combined TOML file.

Reply:
Is there a use case for a normal `Dict`? We don't have any exposed functions which could turn a `Dict` into a `ParamDict`

Reply:
Ahh my mistake, I thought there was already an option to pass 2 dicts into `create_toml_dict`!

Sorry to add to this PRs scope. But I would be happy to see this addressed here. Basically you just need to change this [function](https://github.com/CliMA/CLIMAParameters.jl/blob/1245d78041fd1604c66c009508bfa28d874d6429/src/file_parsing.jl#L386-L402) to accept dicts instead (and remove the TOML.parsefile)

Then add the new function e.g. 
```julia 
function create_toml_dict(
    ::Type{FT};
    override_file::Union{Nothing, String} = nothing,
    default_file::String = joinpath(@__DIR__, "parameters.toml"),
  kwargs...
) where {FT <: AbstractFloat}
default_dict = isnothing(default_file) ? TOML.parsefile("../parameters.toml") : TOML.parsefile(default_file)
overried_dict = isnothing(override_file) ? nothing  : TOML.parsefile(override_file)

return create_toml_dict(::Type{FT}, default = default_dict, override = override_dict; kwargs...)

Reply:
> How about we always output a combined Dict (that can be turned into a "TOML dict"), and if the user requests it via a flag (e.g. an output filepath) we can also write a combined TOML file.

sounds good, but I think we should have them as separate functions, so that each function does a single thing— users can then use the atomic parts as needed.

Reply:
> Is there a use case for a normal Dict? We don't have any exposed functions which could turn a Dict into a ParamDict

Combining with the parsed args, so that we can run the code with command line options.